lux-lib/src/luarc/mod.rs

```diff
@@ -0,0 +1,179 @@
+use crate::lockfile::LocalPackageLockType;
+use crate::lockfile::ProjectLockfile;
+use crate::lockfile::ReadOnly;
+use crate::project::Project;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use std::fs;
+use std::path::PathBuf;
+
+#[derive(Serialize, Deserialize, Default, PartialEq, Debug)]
+#[serde(default)]
+struct LuaRC {
+    #[serde(flatten)] // <-- capture any unknown keys here
+    other: BTreeMap<String, serde_json::Value>,
+
+    #[serde(default)]
+    workspace: Workspace,
+}
+
+#[derive(Serialize, Deserialize, Default, PartialEq, Debug)]
+struct Workspace {
+    #[serde(default)]
+    library: Vec<String>,
+}
+
+pub fn update_luarc(config: &Config) {
+    if !config.generate_luarc() {
+        return; // do nothing
+    }
+    let project = Project::current_or_err().expect("failed to get current project");
+    let lockfile = project.lockfile().expect("should have a lockfile");
+    let luarc_path = project.luarc_path();
+    let dependency_tree = project.tree(config).expect("failed to get project tree");
+    let dependency_tree_root_relative_path = dependency_tree
+        .root()
+        .strip_prefix(project.root())
+        .expect("tree root should be a subpath of project root")
+        .to_path_buf();
+
+    let test_dependency_tree = project
+        .test_tree(config)
+        .expect("failed to get project test tree");
+    let test_dependency_tree_root_relative_path = test_dependency_tree
+        .root()
+        .strip_prefix(project.root())
+        .expect("test tree root should be a subpath of project root")
+        .to_path_buf();
+
+    // read the existing .luarc file or create a new one if it doesn't exist
+    let luarc_content = fs::read_to_string(&luarc_path).unwrap_or_else(|_| String::from("{}"));
+
+    let dependency_dirs = find_dependency_dirs(
+        &lockfile,
+        dependency_tree_root_relative_path,
+        &LocalPackageLockType::Regular,
+    );
+
+    let test_dependency_dirs = find_dependency_dirs(
+        &lockfile,
+        test_dependency_tree_root_relative_path,
+        &LocalPackageLockType::Test,
+    );
+
+    let all_dependecy_dirs: Vec<PathBuf> = dependency_dirs
+        .into_iter()
+        .chain(test_dependency_dirs)
+        // make sure the paths actually exist
+        .filter(|path| fs::exists(path).is_ok_and(|exists| exists))
+        .collect();
+
+    let file = generate_luarc(luarc_content.as_str(), all_dependecy_dirs);
+
+    fs::write(&luarc_path, file)
+        .unwrap_or_else(|_| panic!("failed to write {} file", luarc_path.display()));
+}
+
+fn find_dependency_dirs(
+    lockfile: &ProjectLockfile<ReadOnly>,
+    lux_tree_base_dir: PathBuf,
+    local_package_lock_type: &LocalPackageLockType,
+) -> Vec<PathBuf> {
+    let rocks = lockfile.local_pkg_lock(local_package_lock_type).rocks();
+
+    rocks
+        .iter()
+        .map(|t| lux_tree_base_dir.join(format!("{}-{}@{}/src", t.0, t.1.name(), t.1.version())))
```

Comment:
issue: Hand-rolling the construction of the directory is prone to breakage.
Instead, use `Tree::test_tree` to get the `Tree` for the test dependencies and use
`Tree::entrypoint_layout`[^1] or `Tree::dependency_layout`[^1] to get a `RockLayout` struct, which has a `src: PathBuf` field.

[^1]: You can use `LocalPackageLock::is_entrypoint` to determine which one to use.